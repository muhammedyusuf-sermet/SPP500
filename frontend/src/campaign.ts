export interface ICampaignState {
	Id?: string,
	Name: string,
	Description: string;
	// Monsters: Monster.IMonsterState[];
}
