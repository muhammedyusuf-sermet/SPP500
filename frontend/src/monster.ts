export enum Size {
	Tiny = "Tiny",
	Small = "Small",
	Medium = "Medium",
	Large = "Large",
	Huge = "Huge",
	Gargantuan = "Gargantuan"
}

export enum MonsterType {
	Beast = "Beast",
	Dragon = "Dragon",
	Monstrosity = "Monstrosity",
	Humanoid = "Humanoid",
	Fiend = "Fiend",
	Undead = "Undead",
	Giant = "Giant",
	Elemental = "Elemental",
	SwarmOfTinyBeasts = "SwarmOfTinyBeasts",
	Construct = "Construct",
	Celestial = "Celestial",
	Aberration = "Aberration",
	Fey = "Fey",
	Plant = "Plant",
	Ooze = "Ooze"
}

export enum MonsterRace {
	AnyRace = "AnyRace",
	Devil = "Devil",
	Demon = "Demon",
	Human = "Human",
	Shapechanger = "Shapechanger",
	Goblinoid = "Goblinoid",
	Titan = "Titan",
	Gnoll = "Gnoll",
	Gnome = "Gnome",
	Dwarf = "Dwarf",
	Elf = "Elf",
	Orc = "Orc",
	Kobold = "Kobold",
	Lizardfolk = "Lizardfolk",
	Merfolk = "Merfolk",
	Sahuagin = "Sahuagin",
	Grimlock = "Grimlock"
}

export enum Alignment {
	Unaligned = "Unaligned",
	AnyAlignment = "AnyAlignment",
	AnyNonGoodAlignment = "AnyNonGoodAlignment",
	AnyNonEvilAlignment = "AnyNonEvilAlignment",
	AnyNonLawfulAlignment = "AnyNonLawfulAlignment",
	AnyNonChaoticAlignment = "AnyNonChaoticAlignment",
	AnyGoodAlignment = "AnyGoodAlignment",
	AnyEvilAlignment = "AnyEvilAlignment",
	AnyLawfulAlignment = "AnyLawfulAlignment",
	AnyChaoticAlignment = "AnyChaoticAlignment",
	AnyNeutralAlignment = "AnyNeutralAlignment",
	LawfulGood = "LawfulGood",
	LawfulNeutral = "LawfulNeutral",
	LawfulEvil = "LawfulEvil",
	NeutralGood = "NeutralGood",
	Neutral = "Neutral",
	NeutralEvil = "NeutralEvil",
	ChaoticGood = "ChaoticGood",
	ChaoticNeutral = "ChaoticNeutral",
	ChaoticEvil = "ChaoticEvil"
}

export enum Environment {
	Arctic = "Arctic",
	Coastal = "Coastal",
	Desert = "Desert",
	Forest = "Forest",
	Grassland = "Grassland",
	Hill = "Hill",
	Mountain = "Mountain",
	Swamp = "Swamp",
	Underdark = "Underdark",
	Underwater = "Underwater",
	Urban = "Urban"
}

export enum MonsterAction {
	SpecialAbility = "SpecialAbility",
	Action = "Action",
	LegendaryAction = "LegendaryAction"
}

export interface IMonsterAbilities {
	name: string,
	description: string,
}

export enum TypeOfAction {
	Ranged = "Ranged",
	Melee = "Melee",
}

export interface IMonsterActions {
	name: string,
	description: string,
	usageCount: number,
	legendaryAction: boolean,
	typeOfAction?: TypeOfAction,
	toHit: number,
	reach: number,
	targets: number,
	averagedDamage: number,
	damageDice: number,
	damageConstant: number,

}

export interface IMonsterState {
	Id?: number,
	Name: string,
	Size?: Size | string;
	Type?: MonsterType | string;
	Race?: MonsterRace | string;
	Environment?: Environment | string;
	Alignment?: Alignment | string;

	ArmorClass?: number;
	HitPoints?: number;
	HitPointDistribution?: string;

	Speed?: string;
	Languages?: string;

	DamageVulnerabilities?: string;
	DamageResistances?: string;
	DamageImmunities?: string;
	ConditionImmunities?: string;

	ChallengeRating?: number;

	AbilityScores: IMonsterAbilityScoreState;
	Senses: SenseMap;
	Skills: SkillMap;
	SavingThrows: IMonsterSavingThrowState;
	//Actions: IActionState[];
}

export interface SkillMap {
	[skillName: string]: number|undefined;
}

export interface SenseMap {
	[senseName: string]: number|undefined;
}

export interface IMonsterAbilityScoreState {
	[index:string]: number|undefined;
	Strength?: number;
	Dexterity?: number;
	Constitution?: number;
	Intelligence?: number;
	Wisdom?: number;
	Charisma?: number;
}

export interface IMonsterSavingThrowState {
	[index:string]: number|undefined;
	Strength?: number;
	Dexterity?: number;
	Constitution?: number;
	Intelligence?: number;
	Wisdom?: number;
	Charisma?: number;
}

export interface IActionState {
	Name: string,
	Description: string,
	HitBonus?: number,
	Damage?: string,
	DamageBonus?: number,
	Type?: MonsterAction
}
/*export interface IMonster {
	name: string,
	type: MonsterType,
	alignment: MonsterAlignment,
	size: MonsterSize,
	race: MonsterRace,
	environment: MonsterEnvironment,
	resistance?: string,
	damageImmunity?: string,
	conditionImmunity?: string,
	vulnerability?: string,
	armorClass: number,
	hitPoints?: number,
	hitPointDice?: string,
	hitPointDiceAdd?: number,
	speedLand: number,
	speedSwim?: number,
	strStat: number,
	dexStat: number,
	conStat: number,
	intStat: number,
	wisStat: number,
	chaStat: number,
	strSavingThrow?: number,
	dexSavingThrow?: number,
	conSavingThrow?: number,
	intSavingThrow?: number,
	wisSavingThrow?: number,
	chaSavingThrow?: number,
	skillsAthletics: number,
	skillsAcrobatics?: number,
	skillsSleightOfHand?: number,
	skillsStealth?: number,
	skillsArcana?: number,
	skillsHistory?: number,
	skillsInvestigation?: number,
	skillsNature?: number,
	skillsReligion?: number,
	skillsAnimalHandling?: number,
	skillsInsight?: number,
	skillsMedicine?: number,
	skillsPerception?: number,
	skillsSurvival?: number,
	skillsDeception?: number,
	skillsIntimidation?: number,
	skillsPerformance?: number,
	skillsPersuasion?: number,
	sensesBlindsight?: number,
	sensesDarkvision?: number,
	sensesTremorsense?: number,
	sensesTruesight?: number,
	sensesPassivePerception?: number,
	sensesPassiveInvestigation?: number,
	sensesPassiveInsight?: number,
	languages?: string,
	challengeRating: number,
	experiencePoints: number,
	abilities?: IMonsterAbilities[],
	actions?: IMonsterActions[],
}*/