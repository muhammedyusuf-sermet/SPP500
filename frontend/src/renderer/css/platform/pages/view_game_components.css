.view-game-components-container {
	text-align: center;
}
.page-title {
	padding: 5px 15px;
	display: inline-block;
}

.view-game-components-container .tab-root {
	flex-grow: 1;
}

.view-game-components-container .app-bar {
	position: relative;
}

.view-game-components-container .layout {
	width: auto;
	margin-left: 2%;
	margin-right: 2%;
}

.view-game-components-container .card-grid {
	padding: 8px 0;
}

.view-game-components-container .card {
	height: 100%;
	display: flex;
	flex-direction: column;
}

.view-game-components-container .card-media {
	padding-top: 56.25%; /* 16:9 */
}

.view-game-components-container .card-content {
	flex-grow: 1;
}
