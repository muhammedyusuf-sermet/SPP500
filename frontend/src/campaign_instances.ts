import * as Campaign from "./campaign";

// Dummy array of monsters
import EncounterInstances from './encounter_instances';

let encounterExample: Campaign.ICampaignState;
encounterExample = {
	Id: 0,
	Name: "Campaign",
	Summary: "Bloody hell...",
	Notes: "You gonna die!",
	Encounters: EncounterInstances['EncounterInstances'],
};

let CampaignInstances = new Array();

// Generate a dummy array of encounters with different IDs
for (var i = 0; i < 12; i++) {
	let encounter = {} as Campaign.ICampaignState;
	Object.assign(encounter, encounterExample);
	encounter['Id'] = i;
	CampaignInstances.push(encounter);
}

export default {
    CampaignInstances,
}