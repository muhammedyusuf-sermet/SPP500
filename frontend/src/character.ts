import { ICampaignData } from "./campaign";

export enum CharacterRace {
	Dragonborn = "Dragonborn",
	Dwarf = "Dwarf",
	Elf = "Elf",
	Gnome = "Gnome",
	HalfElf = "HalfElf",
	Halfling = "Halfling",
	HalfOrc = "HalfOrc",
	Human = "Human",
	Tiefling = "Tiefling"
}

export enum CharacterClass {
	Barbarian = "Barbarian",
	Bard = "Bard",
	Cleric = "Cleric",
	Druid = "Druid",
	Fighter = "Fighter",
	Monk = "Monk",
	Paladin = "Paladin",
	Ranger = "Ranger",
	Rogue = "Rogue",
	Sorcerer = "Sorcerer",
	Warlock = "Warlock",
	Wizard = "Wizard"
}

export interface ICharacterData {
	Id?: number;
	Name: string;
	Level?: number;
	Race?: CharacterRace;
	Class?: CharacterClass;
	MaxHealth?: number;
	ArmorClass?: number;
	Notes?: string;

	Campaigns?: ICampaignData[];
}
