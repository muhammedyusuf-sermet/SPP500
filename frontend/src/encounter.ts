import { IMonsterData } from "./monster";

export interface IEncounterData {
	Id?: string,
	Name: string,
	Description: string;
	Monsters: IMonsterData[];
}
