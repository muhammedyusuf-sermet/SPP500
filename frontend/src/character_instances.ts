import { CharacterRace, CharacterClass, ICharacterData } from "./character";


let characterExample: ICharacterData;
characterExample = {
	Id: 1,
	Name: "Character",
	Level: 3,
	Race: CharacterRace.Gnome,
	Class: CharacterClass.Paladin,
	MaxHealth: 23,
	ArmorClass: 12,
	Notes: 'aggressive',

	Campaigns: []
};

export const CharacterInstances: ICharacterData[] = new Array();

// Generate a dummy array of characters with different IDs
for (var i = 0; i < 12; i++) {
	let character = {} as ICharacterData;
	Object.assign(character, characterExample);
	character.Id = i + 1;
	CharacterInstances.push(character);
}