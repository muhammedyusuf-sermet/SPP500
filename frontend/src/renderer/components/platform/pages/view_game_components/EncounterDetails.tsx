import * as React from 'react';
import { Input, Control, Label, Button, Tile, Subtitle } from 'bloomer';
import * as EncounterInterface from '../../../../../encounter';

import 'bulma/css/bulma.css';

export interface IEncounterDetailsState {
	encounter: EncounterInterface.IEncounterState,
}

export class EncounterDetails extends React.Component<any, IEncounterDetailsState> {
	constructor(props: any) {
		super(props);
		this.state = {
			encounter: this.props.encounter,
		}
	}

	isEmptyObject(obj: EncounterInterface.IEncounterState) {
		for(var key in obj) {
			if(obj.hasOwnProperty(key))
				return false;
		}
		return true;
	}

	render() {
		if(!this.isEmptyObject(this.state.encounter)) {
			let encounter = this.state.encounter;
			return (
				<form>
					<h1 className="page-title">Encounter Details</h1>
						<Control>
							<Label>Encounter Name</Label>
							<Input  value={encounter.Name} id="name" label="Encounter Name" readOnly/>
						</Control>
						<Control>
							<Label>Encounter Description</Label>
							<Input  value={encounter.Description} id="name" label="Encounter Description" readOnly/>
						</Control>
						<Tile className="box" isVertical>
							<Subtitle>Monsters</Subtitle>
							{encounter.Monsters.map(monster => (
								<Control key={monster.Id}>
									<Label>{monster.Name} {monster.Id}</Label>
								</Control>
							))}
						</Tile>
					<Button className="button" type="submit" disabled> Edit Encounter </Button>
					<Button className="button" onClick = {this.props.resetParentState}> Return to Game Components </Button>
				</form>
			);
		}
		else{
			return (<div>No encounter is selected.</div>);
		}
	}
}