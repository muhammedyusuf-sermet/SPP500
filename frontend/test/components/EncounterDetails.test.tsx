import * as React from "react"
import { shallow, ShallowWrapper } from 'enzyme';
import {EncounterDetails, IEncounterDetailsState} from '../../src/renderer/components/platform/pages/view_game_components/EncounterDetails';
import * as Encounter from '../../src/encounter';
import * as Monster from "../../src/monster";

describe('Test the Encounter View Details Per Encounter', () => {
	let encounterDetailsInstance: ShallowWrapper<any, IEncounterDetailsState, EncounterDetails>;

	beforeEach(() => {
		let encounterExample: Encounter.IEncounterState;
		encounterExample = {
			Id: "",
			Name: "Encounter",
			Description: "This is a fierce encounter...",
			Monsters: [] as Monster.IMonsterState[],
		};
		encounterDetailsInstance = shallow(<EncounterDetails encounter={encounterExample}/>);
	})

	it('renders without crashing', () => {
		expect(encounterDetailsInstance).toBeDefined();
	});

	it('renders correctly when the page is loaded', () => {
		expect(encounterDetailsInstance).toMatchSnapshot();
	});

	it('should return true if the IEncounterState object is empty', () => {
		let selectedEncounter: Encounter.IEncounterState;
		selectedEncounter = {} as Encounter.IEncounterState;
		let isEmpty = encounterDetailsInstance.instance().isEmptyObject(selectedEncounter);

		expect(isEmpty).toEqual(true);
	});

});
