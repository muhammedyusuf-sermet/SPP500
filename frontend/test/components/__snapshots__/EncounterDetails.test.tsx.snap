// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Encounter View Details Per Encounter renders correctly when the page is loaded 1`] = `
<form>
  <h1
    className="page-title"
  >
    Encounter Details
  </h1>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Encounter Name
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      id="name"
      label="Encounter Name"
      readOnly={true}
      value="Encounter"
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Encounter Description
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      id="name"
      label="Encounter Description"
      readOnly={true}
      value="This is a fierce encounter..."
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Tile)
    className="box"
    isVertical={true}
  >
    <withHelpersModifiers(Subtitle)>
      Monsters
    </withHelpersModifiers(Subtitle)>
  </withHelpersModifiers(Tile)>
  <withHelpersModifiers(Button)
    className="button"
    disabled={true}
    type="submit"
  >
     Edit Encounter 
  </withHelpersModifiers(Button)>
  <withHelpersModifiers(Button)
    className="button"
  >
     Return to Game Components 
  </withHelpersModifiers(Button)>
</form>
`;
